src/gui/screen_printing.cpp

```diff
@@ -429,6 +445,42 @@ static void update_remaining_time(screen_t *screen, time_t rawtime) {
 
     window_set_text(pw->w_etime_value.win.id, array.data());
 }
+
+static void update_end_timestamp(screen_t *screen, timestamp_t *now) {
+
+    auto &array = pw->text_etime;
+    char qmark[2] = "";
+
+    if (marlin_vars()->time_to_end == TIME_TO_END_INVALID) {
+        pw->w_etime_value.color_text = COLOR_VALUE_INVALID;
+        strlcpy(qmark, "?", 2);
```

Comment:
why do you copy the "?" string into qmark?
qmake could have been a `static const char qmark[] = "?"` and you either print qmark to get the "?" on the screen or qmark+1 to print an empty string (this part could be optimized further).

Reply:
commit `187f789`